homeworks/12-mars-viewer-mobx/src/components/imagesBlock/ImagesBlock.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { MouseEventHandler, useCallback } from 'react';
+import { observer } from 'mobx-react-lite';
+import favouritesStore from '../../store/favouritesStore';
+import style from '../imagesContainer/imagesContainer.module.css';
+
+export interface CurImg {
+  curImg: string
+  name: string
+  curId: string,
+  active: boolean
+}
+
+export const ImagesBlock: React.FC<CurImg> = observer(({
+  // eslint-disable-next-line react/prop-types
+  curImg, name, curId, active,
+}) => {
+  const { favourites, addFavourite, deleteFavourite } = favouritesStore;
+
+  const handleClickSvg: MouseEventHandler<SVGSVGElement> = useCallback((event) => {
+    const { id } = event.currentTarget;
+    const result = favourites.filter((favourite) => +favourite === +id).length > 0;
```

Comment:
This logic should be in the model, because:
1. in the future this logic could be used in another place
2. your view level knows too match about structure of your models